src/transbank.c

```diff
@@ -107,6 +107,57 @@ BaseResponse *parse_load_keys_close_response(char *buf)
   return response;
 }
 
+TotalsResponse *parse_get_totals_response(char *buf)
+{
+  TotalsResponse *response = malloc(sizeof(TotalsResponse));
+
+  char *w;
+  int i = 1, l = 0, f = 0;
+  for (int x = i; x < strlen(buf); x++)
+  {
+    if (buf[x] == '|')
+    {
+      w = malloc(l * sizeof(char *));
+      strncpy(w, buf + i, l);
+      w[l] = 0;
+
+      f++;
+      i = x + 1; // Set new position
```

Comment:
if you name `i` something like `word_start_position` you don't need this comment